### PR TITLE
feat(ngx-layout): Add drag and drop to the configurable layout

### PR DESCRIPTION
**Pull request by IbenTesara**

### CODE REVIEW

File: projects/layout/README.md [hunk 1]
```diff
@@ -14,11 +14,15 @@ npm install @studiohyperdrive/ngx-layout
...
+#### 2.1.1 Setup
 
-The idea of these components is that the order of the items within the template.html does not define the order of the items in the DOM. This is especially useful when building a customizable view. The order in which the items will be rendered, depends on either an input or a reactive form control.
+By using content projection, we render our components inside of a `ngx-configurable-layout-item`. Each item requires a `key` as an input, which will be used to match the provided component with the two dimensional array we provide to the `ngx-configurable-layout` component.
+
+This means that the order of rendering is now no longer depended on how you provide the components in the template, but by the two dimensional array provided to the `ngx-configurable-layout` component. This significantly streamlines the process and allows you to easily refactor existing flows. 
+
+To provide the array to the `ngx-configurable-layout` component, you can either provide it through the `keys` input, or use the layout as a form input form `reactive forms`.
```
Comment (WHeirstrate) [inline projects/layout/README.md:25]: ... as a form input form `reactive forms`. => ... as a form input from/for `reactive forms`

File: projects/layout/README.md [hunk 2]
```diff
@@ -32,13 +36,43 @@ import { NgxConfigurableLayoutComponent, NgxConfigurableLayoutItemComponent } fr
 ```
 
 ```html
-<ngx-configurable-layout [keys]="['second-item', 'first-item']">
+<ngx-configurable-layout [keys]="[['second-item', 'first-item'], ['third-item']]">
 	<ngx-configurable-layout-item key="first-item">
-		<p>This is the first item in the template.</p>
+		<p>This is the second item in the template.</p>
```
Comment (WHeirstrate) [inline projects/layout/README.md:41]: This is still the first element in the `template`. It is the second item in the `DOM`, though 🙂

File: projects/layout/src/lib/components/configurable-layout/configurable-layout.component.ts [hunk 3]
```diff
@@ -95,27 +93,31 @@ export class NgxConfigurableLayoutComponent
...
 	/**
-	 * The properties to define the grid in which the items should be displayed. The default
-	 * config is `auto` rows and `1` columns.
+	 * Determines how much space each item takes in the row. By default, this is 'fill'
+	 *
+	 * fill - Will make the items take up the available space
+	 * fit-content - Will make the items take up the space of the provided template
+	 * equal - Will make the items take up equal amount of space throughout the grid
+	 *
+	 * @type {NgxConfigurableLayoutItemSizeOption}
```
Comment (WHeirstrate) [inline projects/layout/src/lib/components/configurable-layout/configurable-layout.component.ts:112]: Shouldn't these tsDoc properties be added to every input, then?

**Review by WHeirstrate — CHANGES_REQUESTED** [review]

File: projects/layout/src/lib/components/configurable-layout/configurable-layout.component.ts (hunk 3 above)
Comment (IbenTesara) [reply projects/layout/src/lib/components/configurable-layout/configurable-layout.component.ts:112]: Nah, forgot to delete it!

**Review by WHeirstrate — APPROVED** [review]

**Pull request merged**